File: Go/climbing_stairs/70.go
package climbingstairs

func climbStairs(n int) int {
	if n == 1 || n == 2 {
		return n
	}
	a := 1
	b := 2
	for i := 3; i < n; i++ {
		temp := b
		b += a
		a = temp
	}
	return a + b
}

File: TypeScript/climbing_stairs/70.ts
function climbStairs(n: number): number {
  if (n === 1 || n === 2) return n;
  let a = 1;
  let b = 2;
  for (let i = 3; i < n; i++) {
    let temp = b;
    b += a;
    a = temp;
  }
  return a + b;
}

export default climbStairs;
